### ISSUE TITLE
SwiftFile error upon read

### ISSUE THREAD

**Issue by kyleabeauchamp** [counsyl/stor#43]
I find that the following example fails on py36 but succeeds on py27.   

```
import stor

obs_full_path = "swift://AUTH/testfile.csv"

f = stor.Path(obs_full_path).open()
print(next(f))
```

**Comment by kyleabeauchamp** [comment]
Below is the exception:

```
    286         "Cached buffer of data read from or to be written to Object Storage"
    287         if self.mode in ('r', 'rb'):
--> 288             return self.stream_cls(self._path.read_object())
    289         elif self.mode in ('w', 'wb'):
    290             return self.stream_cls()

TypeError: initial_value must be str or None, not bytes
```

I suspect the integration tests (test_integration_swift.py) are also failing with similar errors.

**Comment by kyleabeauchamp** [comment]
Possibly (very unlikely?) related issue when retries occur: https://bugs.launchpad.net/python-swiftclient/+bug/1682911

**Comment by kyleabeauchamp** [comment]
So I've been trying to wrap my head around which internal objects and buffers should be bytes versus strings.

I *think* the HTTP response content (https://github.com/counsyl/stor/blob/master/stor/swift.py#L606) is always bytes, right?

**Comment by jtratner** [comment]
That sounds right!

**Comment by kyleabeauchamp** [comment]
A second question: what type should the internal buffer be?  Should the internal buffer match the file open mode (either text or bytes) or should it match the final OBS destination (read_object, i.e., bytes).

It's unclear what the "intent" is here because it seems like the design was done from a py27 perspective.

**Comment by jtratner** [comment]
We should make it similarly pythonic - might make sense to split this into
wrappers to read and write specific to Python 3

**Comment by jtratner** [comment]
Really, the bugs here are in `read_object()` and `write_object()`, which we didn't specify well for Python 3 (because we didn't use it much at the time and didn't think about it).  To decide the "correct" thing to do for those methods, we should look at what the underlying client libraries do and then try to return the same thing. THEN OBSFile should wrap them correctly so that all of that is transparent to the user (i.e., `'r'` should return you text decoded to UTF-8 unless you specify an alternate encoding, `'rb'` returns a bytes wrapper, etc).

As we discussed in person, helpful next step would be to write a small test function that just exercises normal python `open()` in various modes (maybe we can even steal from Python's test suite?) and then confirm that it passes with our file proxies for Python 2, Python 3, and files from swift and from s3.

**Comment by kyleabeauchamp** [comment]
Should be fully resolved as of #54 and possibly earlier.
